functions/src/offers/index.ts

```diff
@@ -0,0 +1,63 @@
+import { validate } from 'class-validator';
+import { firestore } from 'firebase-admin';
+import * as functions from 'firebase-functions';
+import { Change, EventContext } from 'firebase-functions/lib/cloud-functions';
+
+import { IOffer, Offer, OfferStatus } from '../models/offers';
+import DocumentSnapshot = firestore.DocumentSnapshot;
+
+// eslint-disable-next-line no-unused-vars,@typescript-eslint/no-unused-vars
+const queueStatusUpdateTriggers = (change: Change<DocumentSnapshot>, context: EventContext): Promise<void[]> => {
+  const offerBefore = change.before ? change.before.data() as Offer : null;
+  const offerAfter = change.after.data() ? change.after.data() as Offer : null;
+
+  const operations: Promise<void>[] = [];
+
+  // A request has just been accepted -- Update request with new information (CAV, new status)
+  if (offerBefore?.status !== OfferStatus.accepted && offerAfter?.status === OfferStatus.accepted) {
+    /* TODO: Update the request with the current CAV as well as set it's status.
+       This will enable the CAV to access the address of the request.
+       const requestId = context.params.requestId;
+       Use the id to find the request and update it's status and current CAV
+     */
+    operations.push(Promise.resolve());
+  }
+
+  return Promise.all(operations);
+};
+
+const validateOffer = (value: IOffer): Promise<void> => {
+  return validate(Offer.factory(value))
+    .then(() => {
+      return Promise.resolve();
+    });
```

Comment:
Just realised that `validate` doesn't actually reject when it sees invalid data... you need to use `validateOrReject` for that...

```suggestion
  return validateOrReject(Offer.factory(value))
```

Though let's not worry about that right now as i'm working on pulling the model out into its own package anyway.

Reply:
The unit tests pass relying on validate? It throws an error that is caught.